### PR TITLE
trsv implementation

### PR DESCRIPTION
**Pull request by TonyYHsieh**
1. use gemv to implement trsv
2. add trsv_ex function

resolves #___

Summary of proposed changes:
-  some of our users didn't accept to use trsm to implement trsv function.
-  We implement trsv function by using gemv function. We are reference to trsm function and replace gemm by gemv to achieve trsv function which has better performance.
-  We also implement trsv_ex and trsv_ex testcase, all implementation are reference to trsm implementation.

### CODE REVIEW

File: clients/benchmarks/client.cpp [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /* ************************************************************************
- * Copyright 2016 Advanced Micro Devices, Inc.
+ * Copyright 2019 Advanced Micro Devices, Inc.
```
Comment (saadrahim) [inline clients/benchmarks/client.cpp]: I don't know our policy but I thought copyright is supposed to 2016-2019

Comment (TonyYHsieh) [reply clients/benchmarks/client.cpp]: update PR for copywrite format

**Comment by TonyYHsieh** [comment]
> The is no need to add TRSV_EX.
> 
> A 3-month-memory allocation scheme for all of rocBLAS has been designed and it will replace TRSM_EX and TRSV_EX

**Comment by imcarsonliao** [comment]
> > The is no need to add TRSV_EX.
> > A 3-month-memory allocation scheme for all of rocBLAS has been designed and it will replace TRSM_EX and TRSV_EX
> 
> So what should I do for this change??
> Can we keep both TRSV_EX and TRSM_EX before new scheme??

Tony, please remove trsv_ex export API, while internally calling to "ex" function is fine.

**Comment by TonyYHsieh** [comment]
> > > The is no need to add TRSV_EX.
> > > A 3-month-memory allocation scheme for all of rocBLAS has been designed and it will replace TRSM_EX and TRSV_EX
> > 
> > 
> > So what should I do for this change??
> > Can we keep both TRSV_EX and TRSM_EX before new scheme??
> 
> Tony, please remove trsv_ex export API, while internally calling to "ex" function is fine.

Hi imcarsonliao,

I have updated commit to remove trsv_ex API.
Please help to review my update.

Thanks,
Tony.

**Pull request merged**